third_party/wabt/src/interp/jit/jit-backend.cc

```diff
@@ -216,6 +216,49 @@ static void emitBinary(sljit_compiler* compiler, Instruction* instr) {
                  args[0].argw, args[1].arg, args[1].argw);
 }
 
+static void emitUnary(sljit_compiler* compiler, Instruction* instr) {
+  Operand* operand = instr->operands();
+  JITArg args[2];
+
+  for (int i = 0; i < 2; ++i) {
+    operandToArg(operand + i, args[i]);
+  }
+
+  sljit_s32 opcode;
+  sljit_s32 mov = SLJIT_MOV;
```

Comment:
Only needed when mov is needed.

Btw: for immedates the result is a constant. This could be computed ahead and use a simple mov.